src/Lecture2.hs

```diff
@@ -48,7 +50,13 @@ zero, you can stop calculating product and return 0 immediately.
 84
 -}
 lazyProduct :: [Int] -> Int
-lazyProduct = error "TODO"
+lazyProduct = multiply 1
+  where
+    multiply :: Int -> [Int] -> Int
+    multiply acc [] = acc
+    multiply acc (x : xs)
+        | x == 0    = multiply 0 []
+        | otherwise = multiply (acc * x) xs
```

Comment:
Correct solution 👍🏻 Alternatively, you can match on zero directly in the pattern

```suggestion
    multiply _   (0 : _)  = 0
    multiply acc (x : xs) = multiply (acc * x) xs
```